can/interfaces/socketcand/socketcand.py

```diff
@@ -13,12 +13,115 @@
 import socket
 import time
 import traceback
+import urllib.parse as urlparselib
+import xml.etree.ElementTree as ET
 from collections import deque
+from typing import List
 
 import can
 
 log = logging.getLogger(__name__)
 
+DEFAULT_SOCKETCAND_DISCOVERY_ADDRESS = ""
+DEFAULT_SOCKETCAND_DISCOVERY_PORT = 42000
+
+
+def detect_beacon(timeout_ms) -> List[can.typechecking.AutoDetectedConfig]:
```

Comment:
```suggestion
def detect_beacon(timeout_ms: int = 3100) -> List[can.typechecking.AutoDetectedConfig]:
```

I would set the default timeout here.